### PR TITLE
refactor(metadata-instance-editor): Add new feature for instance fields

### PR DESCRIPTION
**Pull request by manishPh**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @manishPh has signed the CLA. Thanks for the pull request!

File: src/features/metadata-instance-fields/messages.js [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+    },
+    customKey: {
+        defaultMessage: 'Key',
+        description: 'Label for the key field for custom metadata',
+        id: 'boxui.metadataInstanceEditor.customKey',
+    },
+    customKeyPlaceholder: {
+        defaultMessage: 'e.g. Order Number',
+        description: 'Placeholder for the key field for custom metadata',
+        id: 'boxui.metadataInstanceEditor.customKeyPlaceholder',
```
Comment (priyajeet) [inline src/features/metadata-instance-fields/messages.js]: Probably change IDs to match new name, metadataInstanceFields

File: src/features/metadata-instance-fields/messages.js [hunk 2]
```diff
@@ -0,0 +1,255 @@
+import { defineMessages } from 'react-intl';
+
+const messages = defineMessages({
+    customTitle: {
+        defaultMessage: 'Custom Metadata',
+        description: 'title of the custom metadata card',
+        id: 'boxui.metadataInstanceEditor.customTitle',
+    },
+    customAdd: {
+        defaultMessage: 'Add',
```
Comment (priyajeet) [inline src/features/metadata-instance-fields/messages.js]: Looks like Add is repeated

File: src/features/metadata-instance-fields/__tests__/__snapshots__/TextField.test.js.snap [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 exports[`features/metadata-instance-editor/fields/TextField should correctly render a number field 1`] = `
 <TextInput
-  className="metadata-instance-editor-field-text"
+  className="metadata-instance-field-text"
```
Comment (priyajeet) [inline src/features/metadata-instance-fields/__tests__/__snapshots__/TextField.test.js.snap]: If class is being changed, use the new bdl- SUIT style

File: src/features/metadata-instance-fields/TextField.scss [hunk 4]
```diff
@@ -1,6 +1,6 @@
-@import '../../../styles/variables';
+@import '../../styles/variables';
 
-.metadata-instance-editor-field-text {
+.metadata-instance-field-text {
```
Comment (priyajeet) [inline src/features/metadata-instance-fields/TextField.scss]: Use new bdl style

File: i18n/en-US.properties [hunk 5]
```diff
@@ -900,6 +888,22 @@ boxui.metadataInstanceEditor.templatesNoResults = No Results
 boxui.metadataInstanceEditor.templatesServerHasNoTemplates = Zero templates
 # Overall title of metadata
 boxui.metadataInstanceEditor.templatesTitle = Templates
+# Label for the add button for custom metadata
+boxui.metadataInstanceFields.customAdd = Add
+# Label for the remove button for custom metadata
+boxui.metadataInstanceFields.customRemove = Remove
+# Text to indicate to select one or more values from drop down list
```
Comment (sahiga) [inline i18n/en-US.properties]: Minor: "dropdown" should be a single word here and on line 899

File: i18n/en-US.properties [hunk 6]
```diff
@@ -900,6 +888,22 @@ boxui.metadataInstanceEditor.templatesNoResults = No Results
...
+boxui.metadataInstanceFields.fieldMultiSelectValue = Select Values
+# Text to indicate no value exists
+boxui.metadataInstanceFields.fieldNoValue = No Value
+# Text to indicate to select a value from drop down list
+boxui.metadataInstanceFields.fieldSelectValue = Select Value
+# Text to indicate to set a date
+boxui.metadataInstanceFields.fieldSetDate = Set Date
+# Text to indicate to set a value
+boxui.metadataInstanceFields.fieldSetValue = Set Value
+# Error message displayed if the template has an field type we dont recognize
```
Comment (sahiga) [inline i18n/en-US.properties]: Typo: "don't"

**Pull request merged**